### PR TITLE
Essentials: Port xamarin/Essentials#1846

### PR DESCRIPTION
**Pull request by dimonovdd**
### Description of Change ###

**Port https://github.com/xamarin/Essentials/pull/1846** 

Using [`Rg.Plugins.Popup`](https://github.com/rotorgames/Rg.Plugins.Popup) and [`Xamarin.Essentials`](https://github.com/xamarin/Essentials) together, I noticed some problems:
- Closing Popups closes pickers and etc., but the tasks remain not finished.
- We don't have the option to specify a UIViewController before opening pickers

This is due not only to `Rg.Plugins.Popup`, but also to any projects that use custom controllers for rendering pages. 

### API Changes ###
Added: 
 
```csharp
public static partial class Platform
{
    public static void Init(Func<UIViewController> getCurrentUIViewController)
}
```

### Behavioral Changes ###
- All tasks will be finished when the controller is closed
- Ability to change the parent controller for pickers

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [x] I'm not sure, please help me

### CODE REVIEW

**Review by Redth — APPROVED** [review]

**Comment by mattleibow** [comment]
This feels good. Just going to test on my new iPhone and will see if it is a merge today :)

File: src/Essentials/src/Platform/Platform.ios.tvos.watchos.cs [hunk 1]
```diff
@@ -138,5 +146,24 @@ internal static UIWindow GetCurrentWindow(bool throwIfNull = true)
...
+#if __IOS__
+		internal class UIPresentationControllerDelegate : UIAdaptivePresentationControllerDelegate
+		{
+			readonly Action dismissHandler;
+
+			internal UIPresentationControllerDelegate(Action dismissHandler)
+				=> this.dismissHandler = dismissHandler;
+
+			public override void DidDismiss(UIPresentationController presentationController) =>
+				dismissHandler?.Invoke();
```
Comment (mattleibow) [inline src/Essentials/src/Platform/Platform.ios.tvos.watchos.cs:159]: Can we set this to null ofter invoking? This way we don't get a double finish. Same with the Dispose invocation.

Comment (dimonovdd) [reply src/Essentials/src/Platform/Platform.ios.tvos.watchos.cs]: yes, we can do it

**Review by mattleibow — APPROVED** [review]

**Pull request merged**